src/braket/circuits/noises.py

```diff
@@ -1349,27 +1318,32 @@ def from_dict(cls, noise: dict) -> Noise:
 Noise.register_noise(Kraus)
 
 
-def _ascii_representation(noise: str, parameters: List[Union[FreeParameter, float]]) -> str:
+def _ascii_representation(
+    noise: str, parameters: List[Union[FreeParameterExpression, float]]
+) -> str:
     """
     Generates a formatted ascii representation of a noise.
 
     Args:
         noise (str): The name of the noise.
-        parameters (List[Union[FreeParameter, float]]): The parameters to the noise.
+        parameters (List[Union[FreeParameterExpression, float]]): The parameters to the noise.
 
     Returns:
-        str: Returns the ascii representation for the noise.
-
+        str: The ascii representation of the noise.
     """
     param_list = []
     for param in parameters:
         param_list.append(
-            str(param) if isinstance(param, FreeParameter) else "{:.2g}".format(param)
+            str(param) if isinstance(param, FreeParameterExpression) else "{:.2g}".format(param)
         )
     param_str = ",".join(param_list)
     return f"{noise}({param_str})"
 
 
+def _substitute_value(expr, **kwargs):
```

Comment:
Do we check whether the result is within bounds before submitting the circuit to a device?

Reply:
Since `bind_values` instantiates a new object, parameter substitution is automatically checked.